src/Foundation/NSMutableDictionary.cs

```diff
@@ -335,13 +336,20 @@ public void LowlevelSetObject (IntPtr obj, IntPtr key)
 
 		public void LowlevelSetObject (NSObject obj, IntPtr key)
 		{
-			if (obj == null)
-				throw new ArgumentNullException (nameof (obj));
-#if MONOMAC
-			ObjCRuntime.Messaging.void_objc_msgSend_IntPtr_IntPtr (this.Handle, selSetObject_ForKey_Handle, obj.Handle, key);
-#else
-			ObjCRuntime.Messaging.void_objc_msgSend_IntPtr_IntPtr (this.Handle, Selector.GetHandle ("setObject:forKey:"), obj.Handle, key);
-#endif
+			if (obj is null)
+				ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (obj));
+
+			LowlevelSetObject (obj.Handle, key);
+		}
+
+		public void LowlevelSetObject (string obj, IntPtr key)
```

Comment:
So named parameters are easier

```suggestion
		public void LowlevelSetObject (string str, IntPtr key)
```